libnetwork/driverapi/endpoint.go

```diff
@@ -0,0 +1,26 @@
+package driverapi
+
+// EndpointDriver is a driver managing endpoints.
+type EndpointDriver interface {
+	Driver
```

Comment:
In all situations where this is used, it must implement _both_ the `Driver` and `Endpoint...` methods or could there be cases where _only_ the moved methods are needed? Mostly considering here if the methods that were moved should be their own interface, and `EndPointDriver` to be a composite, like;

```go
// EndpointDriver is a driver managing endpoints.
type EndpointDriver interface {
	Driver
	EndPointManager
}
```

Or even if `Join` and `Leave` are somewhat separate from that;

```go
// EndpointDriver is a driver managing endpoints.
type EndpointDriver interface {
	Driver
	EndPointManager
	SandboxManager
}
```

Reply:
As far as now, there's no 'network-less' endpoint driver, so there's no need to create a dedicated `EndpointManager` interface.

Reply:
Gotcha; yup, makes sense